sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -406,6 +406,150 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
             raise ValueError("Unsupported BlobType: {}".format(blob_type))
         return kwargs
 
+    def _upload_blob_from_url_options(self, source_url, **kwargs):
+        # type: (...) -> Dict[str, Any]
+        headers = kwargs.pop('headers', {})
+        tier = kwargs.pop('standard_blob_tier', None)
+        include_source_blob_properties = kwargs.pop('include_source_blob_properties', True)
+        timeout = kwargs.pop('timeout', None)
+        dest_mod_conditions = get_modify_conditions(kwargs)
+        blob_tags_string = serialize_blob_tags_header(kwargs.pop('tags', None))
+        content_settings = kwargs.pop('content_settings', None)
+        source_content_md5 = kwargs.pop('source_content_md5', None)
+        if content_settings:
+            kwargs['blob_http_headers'] = BlobHTTPHeaders(
+                blob_cache_control=content_settings.cache_control,
+                blob_content_type=content_settings.content_type,
+                blob_content_md5=None,
+                blob_content_encoding=content_settings.content_encoding,
+                blob_content_language=content_settings.content_language,
+                blob_content_disposition=content_settings.content_disposition
+            )
+        cpk = kwargs.pop('cpk', None)
+        cpk_info = None
+        if cpk:
+            if self.scheme.lower() != 'https':
+                raise ValueError("Customer provided encryption key must be used over HTTPS.")
+            cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
+                               encryption_algorithm=cpk.algorithm)
+        kwargs['cpk_info'] = cpk_info
+        kwargs['cpk_scope_info'] = get_cpk_scope_info(kwargs)
+
+        options = {
+            'content_length': 0,
+            'copy_source_blob_properties': include_source_blob_properties,
+            'source_content_md5': bytearray(source_content_md5) if source_content_md5 else None,
+            'copy_source': source_url,
```

Comment:
source_url needs to be url encoded?

Reply:
It has been encoded here https://github.com/Azure/azure-sdk-for-python/blob/fdd2f5f4dd933c7abbd49e972a7f1efec9a93e3e/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L546.